### PR TITLE
FIRECERT-2143: [FCS] Implement Setter/Getter 'set' Glue Code

### PR DESCRIPTION
**Pull request by Abhishk123**

### CODE REVIEW

File: cypress/support/cypress-commands/commands.js [hunk 1]
```diff
@@ -35,15 +35,25 @@ Cypress.Commands.add(
...
     // Reading the data from combinedJson based on key.
     let fireboltData;
+    let logMessage = CONSTANTS.NO_DATA_FOR_THE_KEY + key;
     if (callType == CONSTANTS.SUPPORTED_CALLTYPES.FIREBOLTMOCKS) {
       fireboltData = UTILS.getEnvVariable(CONSTANTS.COMBINEDFIREBOLTMOCKS)[key];
     } else if (callType == CONSTANTS.SUPPORTED_CALLTYPES.SET_RESPONSE_JSON) {
       fireboltData = UTILS.getEnvVariable('setResponseJson')[key];
     } else {
-      fireboltData = UTILS.getEnvVariable(CONSTANTS.COMBINEDFIREBOLTCALLS)[key];
+      if (key) {
```
Comment (jbigel) [inline cypress/support/cypress-commands/commands.js:44]: I think this logic should be apart of the glue command. We don't want to set this runtime variable for non setter/getter calls.

Comment (Abhishk123) [reply cypress/support/cypress-commands/commands.js]: removed this changes from here and added in glue itself. will move it to some function in optimization ticket

File: cypress/support/step_definitions/fireboltCalls.js [hunk 2]
```diff
@@ -422,3 +422,141 @@ Given(/User triggers event with value as '(.+)'/, (key) => {
...
+ * @function And '(.+)' invokes the '(.+)' API '(.+)' to set '(.+)' to '(.+)'
+ * @description send message to platform to make api call.
+ * @param {String} appId - app identifier, determines for which App sending the message to make a API call.
+ * @param {String} sdk - sdk name.
+ * @param {String} fireboltCallKey - key name passed to look for firebolt call object in fireboltCallData.
+ * @param {String} attribute - The attribute we are setting (ex. fontFamily).
+ * @param {String} value - The value we set the attribute to (ex. monospaced_sanserif)
+ * @example
+ * Given '1st party app' invokes the 'Firebolt' API 'CLOSEDCAPTION_SETTINGS' to set 'enable' to 'true'
+ * Given '3rd party app' invokes the 'Firebolt' API 'CLOSEDCAPTION_SETTINGS' to set 'enable' to 'true'
```
Comment (jbigel) [inline cypress/support/step_definitions/fireboltCalls.js:437]: 1st party update

File: cypress/support/step_definitions/fireboltCalls.js [hunk 3]
```diff
@@ -422,3 +422,141 @@ Given(/User triggers event with value as '(.+)'/, (key) => {
...
+          UTILS.getEnvVariable('runtime').fireboltCall = fireboltData;
+        });
+      } else {
+        fireboltCallObject = UTILS.getEnvVariable('runtime').fireboltCall;
+        fireboltCallObjectErrorMessage =
+          'Unable to find the firebolt object in the runtime environment variable';
+      }
+
+      cy.then(() => {
+        if (!fireboltCallObject) {
```
Comment (jbigel) [inline cypress/support/step_definitions/fireboltCalls.js:464]: Can you add some descriptive comments

**Review by jbigel — CHANGES_REQUESTED** [review]

File: cypress/support/step_definitions/fireboltCalls.js [hunk 4]
```diff
@@ -422,3 +422,141 @@ Given(/User triggers event with value as '(.+)'/, (key) => {
     throw new Error(CONSTANTS.STEP_IMPLEMENTATION_MISSING);
   });
 });
+
+/**
+ * @module fireboltCalls
+ * @function And '(.+)' invokes the '(.+)' API '(.+)' to set '(.+)' to '(.+)'
```
[the remark this answers is not in the document]
Comment (Abhishk123) [reply cypress/support/step_definitions/fireboltCalls.js:428]: addressed the comment

File: cypress/support/step_definitions/fireboltCalls.js (hunk 2 above)
Comment (Abhishk123) [reply cypress/support/step_definitions/fireboltCalls.js:437]: addressed the comment

File: cypress/support/step_definitions/fireboltCalls.js (hunk 3 above)
Comment (Abhishk123) [reply cypress/support/step_definitions/fireboltCalls.js:464]: addressed the comment

File: cypress/support/cypress-support/src/utils.js [hunk 5]
```diff
@@ -741,6 +741,38 @@ class FireLog {
...
+ *
+ * @returns
+ * 123
+ * true
+ */
+function parseValue(str) {
+  if (str === null || str === undefined) return str;
+
+  if (typeof str === 'string') {
+    if (str === 'true') return true;
```
[the remark this answers is not in the document]
Comment (Abhishk123) [reply cypress/support/cypress-support/src/utils.js:762]: I don't think we use boolean value as string ex: "true", we will use boolean value always. In negative scenario we will use random string, something like "test"

[the remark this answers is not in the document]
Comment (neeradanelxsi) [reply cypress/support/cypress-support/src/utils.js:762]: yes, we don't have a use case where we send **true** as a string.

**Review by jbigel — APPROVED** [review]

**Pull request merged**